core/server/services/auth/api-key/admin.js

```diff
@@ -0,0 +1,82 @@
+const jwt = require('jsonwebtoken');
+const models = require('../../../models');
+const {
+    InternalServerError,
+    UnauthorizedError
+} = require('../../../lib/common/errors');
+
+const JWT_OPTIONS = {
+    maxAge: '5m',
+    algorithms: ['HS256']
+};
+
+/**
+ * Remove 'Bearer' from raw authorization header and extract the API Key ID and
+ * JWT token. Eg. Authorization: Bearer ${objectId}:${JWT}
+ * @param {string} header
+ */
+const extractCredentialsFromHeader = function extractCredentialsFromHeader(header) {
+    let [scheme, credentials] = header.split(' ');
+
+    if (/^Bearer$/i.test(scheme)) {
+        let [apiKeyId, token] = credentials.split('|');
+
+        return {
+            apiKeyId,
+            token
+        };
+    } else {
+        return {};
+    }
+};
+
+const authenticateAdminAPIKey = function authenticateAdminAPIKey(req, res, next) {
+    // allow fallthrough to other auth methods or final ensureAuthenticated check
+    if (!req.headers || (req.headers && !req.headers.authorization)) {
+        return next();
+    }
+
+    let {apiKeyId, token} = extractCredentialsFromHeader(req.headers.authorization);
+
+    if (!apiKeyId || !token) {
+        return next(new UnauthorizedError({
+            message: 'Authorization header format is "Authorization: Bearer [api key id]|[token]"',
+            code: 'INVALID_AUTH_HEADER'
+        }));
+    }
+
+    models.ApiKey.findOne({id: apiKeyId}).then((apiKey) => {
+        if (!apiKey) {
+            return next(new UnauthorizedError({
+                message: 'Unknown API Key',
+                code: 'UNKNOWN_API_KEY'
+            }));
+        }
+
+        // TODO: should we do Buffer.from(x, 'hex') or is using the secret as-is ok?
+        try {
+            // TODO: grab the decoded payload and check if the payload endpoint
+            // matches the requested endpoint
+            jwt.verify(token, apiKey.get('secret'), JWT_OPTIONS);
```

Comment:
@allouis is there anything else we want to do with the JWT payload?

Reply:
I think we should add an `audience` option to `JWT_OPTIONS`

Reply:
I also think we should be using the `keyid` header claim or the `issuer` to get the id for the key as this is the standard way to do it.

Reply:
What would the value of the `audience` option be?

Reply:
It would be unique identifier of the server - usually it's a URL so `config.get('url') + /ghost/admin` I would think.

Reply:
What does that buy us and does it have any effect on how the SDK creates tokens?

Reply:
The main thing is that it keeps us spec conformant (https://tools.ietf.org/html/rfc7519#section-4.1.3)
> Each principal intended to process the JWT MUST identify itself with a value in the audience claim.

But it also gives us a simple way to do endpoint scoped tokens we spoke about before :)

As for affect on the SDK, that will just have to pass `audience` param too, which i guess for now will be `apiHost` or something similar?

Reply:
FYI
audience : should be the receiver of the token
issuer : should be the generator/creator of the token (ie the ghost server-s app url)
check the oauth spec &/ google jwt claims :)